### PR TITLE
Remove more obsolete statistics

### PR DESCRIPTION
**Pull request by ltamasi**
Test Plan:
`make check`

### CODE REVIEW

**Comment by siying** [comment]
So the decision is to give up renumbering stats and leave holes in the middle. Just to confirm it is the case.

File: include/rocksdb/statistics.h [hunk 1]
```diff
@@ -147,24 +141,19 @@ enum Tickers : uint32_t {
...
-  RATE_LIMIT_DELAY_MILLIS,
 
   // Number of MultiGet calls, keys read, and bytes read
   NUMBER_MULTIGET_CALLS,
   NUMBER_MULTIGET_KEYS_READ,
   NUMBER_MULTIGET_BYTES_READ,
 
-  // Number of deletes records that were not required to be
-  // written to storage because key does not exist
-  NUMBER_FILTERED_DELETES,
```
Comment (pdillinger) [inline include/rocksdb/statistics.h:167]: Small chance of bringing this feature back, but OK

File: include/rocksdb/statistics.h [hunk 2]
```diff
@@ -464,11 +451,6 @@ enum Histograms : uint32_t {
   READ_BLOCK_COMPACTION_MICROS,
   READ_BLOCK_GET_MICROS,
   WRITE_RAW_BLOCK_MICROS,
-  STALL_L0_SLOWDOWN_COUNT,
-  STALL_MEMTABLE_COMPACTION_COUNT,
-  STALL_L0_NUM_FILES_COUNT,
-  HARD_RATE_LIMIT_DELAY_COUNT,
-  SOFT_RATE_LIMIT_DELAY_COUNT,
```
Comment (pdillinger) [inline include/rocksdb/statistics.h:471]: Not sure if we *should* have good statistics for these things, but it's clear we *don't*

**Review by pdillinger — APPROVED** [review]
LGTM

**Comment by pdillinger** [comment]
I think the renumbering of the C++ enum should have been noted in HISTORY for #11123 (and could be remedied here). Warn as a public API change--not just removal--in case people are relying on casted ordinal values of these stats.

> So the decision is to give up renumbering stats and leave holes in the middle. Just to confirm it is the case.

I think you're talking about Java. AFAIK, that's the best handling until we run out of values.

**Comment by ltamasi** [comment]
> So the decision is to give up renumbering stats and leave holes in the middle. Just to confirm it is the case.

Right. Based on my research, the current conventions (or lack thereof) are:
* C++ ids are not guaranteed to be stable. They have shifted around many times as people routinely added new tickers in the middle of the list.
* On the contrary, Java ids are meant to be stable; new tickers are always added to the end of the list.
* As a consequence, Java and C++ ids are not guaranteed to match. In addition to the same item potentially appearing at a different position, or completely missing from the Java API, even the data types are different: in C++, we have `uint32_t`, while in Java we have a signed byte (which actually leads to negative ids for several newer items).

**Comment by pdillinger** [comment]
Perhaps in follow-up we can adjust this comment:

>  *  1. Any ticker should be added immediately before TICKER_ENUM_MAX, taking
>  *     over its old value.

And re-arrange some of the tickers being kept to have better logical grouping. :)

**Comment by ltamasi** [comment]
Yeah, good idea. Actually, let me add update the comment to explicitly say that C++ enum values are not guaranteed to be stable.

**Pull request closed without merging**